gateway/uplink.go

```diff
@@ -47,7 +47,12 @@ func (g *Gateway) parseDataUplink(ctx context.Context, phyPayload []byte) (strin
 	// frame port specifies application port - 0 is for MAC commands 1-255 for device messages.
 	fPort := phyPayload[8+foptsLength]
 
-	// device data in the message.
+	// Ensure there is a frame payload in the packet.
+	if int(8+foptsLength+1) >= (len(phyPayload) - 4) {
+		return "", map[string]interface{}{}, errors.New("device sent packet with no data")
```

Comment:
[opt] maybe add device name, though after talking with olivia it sounded like there wasn't much a user could do for this

Reply:
added the name, not really anything the user can do to fix this but still may be helpful to know which device is doing this.